<html>

<head>
<title>Opening a FastQ file</title>
<style type="text/css">
	body {
		font-family: sans-serif;
	}
</style>
</head>
<body>
<h1>Opening a Sequence file</h1>
<p>
To open one or more Sequence files interactively simply run the
program and select File &gt; Open.  You can then select the files
you want to analyse.
 </p>
 <p>
 Newly opened files will immediately appear in the set of tabs
 at the top of the screen.  Because of the size of these files
 it can take a couple of minutes to open them.  FastQC operates
 a queueing system where only one file is opened at a time, and
 new files will wait until existing files have been processed.
 </p>
 <p>
 FastQC supports files in the following formats
 </p>
 <ul>
 <li>FastQ (all quality encoding variants)</li>
 <li>Casava FastQ files*</li>
 <li>Colorspace FastQ</li>
 <li>GZip compressed FastQ</li>
 <li>SAM</li>
 <li>BAM</li>
 <li>SAM/BAM Mapped only (normally used for colorspace data)</li>
 </ul>
 
 <p>
 * Casava fastq format is the same as regular fastq except that
 the data is usually split across multiple files for a single sample.
 In this mode the program will merge the files in a sample group and
 present a single report for each sample. Also Casava fastq files
 contain poor quality sequences which have been flagged to be remove.
 In Casava mode the program will exclude these flagged sequences from
 the report.
 </p>
 
 <p>
 By default FastQC will try to guess the file format from the name
 of the input file.  Anything ending in .sam or .bam will be 
 opened as a SAM/BAM file (using all sequences, mapped and unmapped)
 , and everything else will be treated as FastQ format.  If you want 
 to override this detection and specify the file format manually
 then you can use the drop down file filter in the file chooser to
 select the type of file you're going to load. You need to use the 
 drop down selector to make the program use the Mapped BAM or Casava
 file modes as these won't be selected automatically.
 </p>
 
 
</body>

</html>